storage/src/main/java/tech/pegasys/teku/storage/server/VersionedDatabaseFactory.java

```diff
@@ -91,9 +128,16 @@ private void createDirectories() {
               "Unable to create the path to store database files at %s",
               dbDirectory.getAbsolutePath()));
     }
+    if (!archiveDirectory.exists() && !archiveDirectory.mkdirs()) {
```

Comment:
hmm, this (and the one above) should actually be:
```suggestion
    if (!archiveDirectory.mkdirs() && !archiveDirectory.isDirectory()) {
```

ie try to make it and be happy as long as we wind up with a directory.  It's unlikely to matter but this way handles weird corner cases and race conditions better - plus gives a useful error message if it happens to exist as a file instead of a directory.